src/overlays/actors/ovl_En_Giant/z_en_giant.c

```diff
@@ -26,42 +31,449 @@ const ActorInit En_Giant_InitVars = {
+extern AnimationHeader D_0600A1C4;
+extern AnimationHeader D_0600ACA4;
+extern AnimationHeader D_0600B784;
+extern AnimationHeader D_0600C5D4;
+extern AnimationHeader D_0600D040;
+extern AnimationHeader D_0600DE84;
+extern AnimationHeader D_060102A4;
+extern AnimationHeader D_060116E4;
+extern AnimationHeader D_06012A38;
+extern AnimationHeader D_06013004;
+extern AnimationHeader D_06013FE8;
+extern AnimationHeader D_06015334;
+extern AnimationHeader D_06017944;
+
+static AnimationHeader* sAnimationTable[] = {
+    &D_06008394, &D_060096E4, &D_060102A4, &D_060116E4, &D_06012A38, &D_06013004, &D_06013FE8, &D_06015334,
+    &D_06017944, &D_0600A1C4, &D_0600D040, &D_0600DE84, &D_0600ACA4, &D_0600B784, &D_0600C5D4,
+};
+
+static TexturePtr sFaceTextures[] = { D_06005A80, D_06006280, D_06006A80 };
+
+void EnGiant_ChangeAnimation(EnGiant* this, s16 newAnimationId) {
+    if (newAnimationId >= GIANT_ANIMATION_LOOK_UP_START && newAnimationId < GIANT_ANIMATION_MAX) {
+        if ((this->animationId == GIANT_ANIMATION_WALKING_LOOP && newAnimationId != GIANT_ANIMATION_WALKING_LOOP) ||
+            (newAnimationId == GIANT_ANIMATION_WALKING_LOOP && this->animationId != GIANT_ANIMATION_WALKING_LOOP)) {
+            SkelAnime_ChangeAnim(&this->skelAnime, sAnimationTable[newAnimationId], 1.0f, 0.0f,
+                                 SkelAnime_GetFrameCount(&sAnimationTable[newAnimationId]->common), 2, 10.0f);
+        } else {
+            SkelAnime_ChangeAnimDefaultStop(&this->skelAnime, sAnimationTable[newAnimationId]);
+        }
+        this->animationId = newAnimationId;
+    }
+}
+
+s32 EnGiant_IsNotFreed(EnGiant* this) {
+    switch (GIANT_TYPE(&this->actor)) {
+        case GIANT_TYPE_SWAMP_TERMINA_FIELD:
+        case GIANT_TYPE_SWAMP_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_SWAMP_GIANTS_CHAMBER_AND_ENDING:
+        case GIANT_TYPE_SWAMP_CLOCK_TOWER_FAILURE:
+            if (!CHECK_QUEST_ITEM(0)) {
+                return 1;
+            }
+            break;
+        case GIANT_TYPE_MOUNTAIN_TERMINA_FIELD:
+        case GIANT_TYPE_MOUNTAIN_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_MOUNTAIN_GIANTS_CHAMBER_AND_ENDING:
+        case GIANT_TYPE_MOUNTAIN_CLOCK_TOWER_FAILURE:
+            if (!CHECK_QUEST_ITEM(1)) {
+                return 1;
+            }
+            break;
+        case GIANT_TYPE_OCEAN_TERMINA_FIELD:
+        case GIANT_TYPE_OCEAN_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_OCEAN_GIANTS_CHAMBER_AND_ENDING:
+        case GIANT_TYPE_OCEAN_CLOCK_TOWER_FAILURE:
+            if (!CHECK_QUEST_ITEM(2)) {
+                return 1;
+            }
+            break;
+        case GIANT_TYPE_CANYON_TERMINA_FIELD:
+        case GIANT_TYPE_CANYON_CLOCK_TOWER_SUCCESS:
+        case GIANT_TYPE_CANYON_GIANTS_CHAMBER_AND_ENDING:
+        case GIANT_TYPE_CANYON_CLOCK_TOWER_FAILURE:
+            if (!CHECK_QUEST_ITEM(3)) {
+                return 1;
+            }
+            break;
+    }
+
+    return 0;
+}
```

Comment:
```suggestion
            if (!CHECK_QUEST_ITEM(0)) {
                return true;
            }
            break;
        case GIANT_TYPE_MOUNTAIN_TERMINA_FIELD:
        case GIANT_TYPE_MOUNTAIN_CLOCK_TOWER_SUCCESS:
        case GIANT_TYPE_MOUNTAIN_GIANTS_CHAMBER_AND_ENDING:
        case GIANT_TYPE_MOUNTAIN_CLOCK_TOWER_FAILURE:
            if (!CHECK_QUEST_ITEM(1)) {
                return true;
            }
            break;
        case GIANT_TYPE_OCEAN_TERMINA_FIELD:
        case GIANT_TYPE_OCEAN_CLOCK_TOWER_SUCCESS:
        case GIANT_TYPE_OCEAN_GIANTS_CHAMBER_AND_ENDING:
        case GIANT_TYPE_OCEAN_CLOCK_TOWER_FAILURE:
            if (!CHECK_QUEST_ITEM(2)) {
                return true;
            }
            break;
        case GIANT_TYPE_CANYON_TERMINA_FIELD:
        case GIANT_TYPE_CANYON_CLOCK_TOWER_SUCCESS:
        case GIANT_TYPE_CANYON_GIANTS_CHAMBER_AND_ENDING:
        case GIANT_TYPE_CANYON_CLOCK_TOWER_FAILURE:
            if (!CHECK_QUEST_ITEM(3)) {
                return true;
            }
            break;
    }

    return false;
}
```
This is returning a boolean. Love the case labels. You can add the quest item enums, no need to wait for @engineer124's PR, provided we agree on what they should be...

Reply:
Do you mean grab the whole `QuestItem` enum and put it in z64item.h? I can do that, and I agree on what the values are, but I just wanted to make sure I'm doing the right thing and not stepping on another programmer's toes.